backend-elasticsearch/src/main/java/org/hibernate/search/v6poc/backend/elasticsearch/types/dsl/impl/IntegerIndexSchemaFieldContext.java

```diff
@@ -42,8 +40,14 @@ protected PropertyMapping contribute(IndexSchemaFieldDefinitionHelper<Integer> h
 			ElasticsearchIndexSchemaObjectNode parentNode) {
 		PropertyMapping mapping = super.contribute( helper, collector, parentNode );
 
+		StandardFieldConverter<Integer> converter = new StandardFieldConverter<>(
+				helper.createUserIndexFieldConverter(),
+				IntegerFieldCodec.INSTANCE
+		);
+		StandardFieldPredicateBuilderFactory predicateBuilderFactory =
+				new StandardFieldPredicateBuilderFactory( converter );
```

Comment:
I wonder if it could be possible to not create a new instance every time in the simple case. But I'm not sure you can be aware of when it is the simple case.